artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/converter/CoreAmqpConverter.java

```diff
@@ -173,9 +174,12 @@ public static AMQPMessage fromCore(ICoreMessage coreMessage) throws Exception {
          properties.setTo(toAddress(destination));
          maMap.put(JMS_DEST_TYPE_MSG_ANNOTATION, destinationType(destination));
       }
-      Destination replyTo = message.getJMSReplyTo();
```

Comment:
Why remove this seemed a cleaner way to encapsulate

Reply:
This was using a fake (incomplete) JMS Layer only available on Server, that only had ActiveMQDestination as base, and in the end it was just to the the property.

Reply:
you would need to implement the other 4 types of classes on the JMS Layer, to get to the same result in the end.

Reply:
why not? would make it cleaner. Out of query why did we create a JMS representation of amqp qpid bits, when actually already qpid-jms has done this. Then it would make JMS to JMS much cleaner, as you would parse it through the same mapping logic to handle any foreign JMS message when you do a JMS produce.

Reply:
@clebertsuconic unresolving - as this is the other query im referring to from the openwire class.

Reply:
not possible. This is just setting the properties directly.

Reply:
even so, it would be duplication. Those JMS Client classes are not available at the server's. It would be duplication anyways.

This is just at the conversion side for this property.

Reply:
fair enough as was a query more than anything.